fintual-oss/paraffin-back
# Feat learning unit successions model

# Context
- With this PR, we are creating a LearningUnitSuccession model, which is the relation of precedence between learning units in a cycle.

# Changelog
- Create the model with rails validations 
- Create the migration with same validations at a database level
- Updated the schema
- Create the factory for LearningUnitSuccession creation
- Create the model spec, testing both the model validations and the factory valid build 
- Rename 2 factory files that were not following the naming conventions
- Add .env file to .gitignore

# QA
- Migrate your database
- Launch Rails console
- Create a LearningUnitSuccession instance, with both valid and invalid attributes
- You should be able to create an entry in the database only if it has a valid cycle and 2 valid learning units, differents one from the other (predecessor and successor), associated.
